README.md

```diff
@@ -102,12 +103,39 @@ In a typical usage scenario you'll want to call one of the `Tracer.getInstance()
 
 The `extractParentSpanFromRequest()` method is potentially different for different applications, however for HTTP-based frameworks the pattern is usually the same - look for and extract distributed-tracing-related information from the HTTP headers and use that information to create a parent span. There is a utility method that performs this work for you: `HttpRequestTracingUtils.fromRequestWithHeaders(RequestWithHeaders, List<String>)`. You simply need to provide the HTTP request wrapped by an implementation of `RequestWithHeaders` and the list of user ID header keys for your application (if any) and it will do the rest using the standard distributed tracing header key constants found in `TraceHeaders`. See the javadocs for those classes and methods for more information and usage instructions.
 
-**NOTE:** Given the thread-local nature of this library you'll want to make sure the span completion call is in a finally block or otherwise guaranteed to be called no matter what (even if the request fails with an error) to prevent problems when subsequent requests are processed on the same thread. The `Tracer` class does its best to recover from incorrect thread usage scenarios and log information about what happened but the best solution is to prevent the problems from occurring in the first place.
- 
+**NOTE:** Given the thread-local nature of this library you'll want to make sure the span completion call is in a finally block or otherwise guaranteed to be called no matter what (even if the request fails with an error) to prevent problems when subsequent requests are processed on the same thread. The `Tracer` class does its best to recover from incorrect thread usage scenarios and log information about what happened but the best solution is to prevent the problems from occurring in the first place. See the section below on `try-with-resources` for some tips on foolproof ways to safely complete your spans.
+
 <a name="servlet_filter_info"></a>  
 #### Is your application running in a Servlet-based framework?
 
 If your application is running in a Servlet environment (e.g. Spring MVC, Jersey, raw Servlets, etc) then this entire lifecycle can be handled by a Servlet `Filter`. We've created one for you that's ready to drop in and go - see the [wingtips-servlet-api](wingtips-servlet-api/README.md) Wingtips plugin module library for details. That plugin module is also a good resource to see how the code for a production-ready implementation of this library might look.
+
```

Comment:
I like how you made Closeable, that's a nice change!

Could we also add an API for starting a span in the context of the current thread as well?  It would be a companion to startRequestWithRootSpan and startSubSpan that would call one or the other based on the current state of the span stack.  Something like,

``` java
public Span startSpanInContext(String spanName)
```

And the language you could use when introducing the API could be something like "when you want to start a span in your function but don't control the context under which your function is called (e.g. you are providing a library), use this method to start a Span - it will call startRequestWithRootSpan or startSubSpan as appropriate based on the state of the Span stack for the calling thread at the time of the call."  I just think for the safety of library authors, we need to provide this convenience method.

Reply:
Normally I'd argue against it as you should always know whether you're starting an overall request span or a subspan based on your current context, and the errors that pop up in the logs help you figure out when you're using Wingtips incorrectly. But this is a use case I hadn't thought of, and I like the way you phrase it. I think I'll add your suggestion to this PR. Thank you!